activerecord/test/cases/relation_test.rb

```diff
@@ -363,6 +363,12 @@ def test_relation_with_optimizer_hints_filters_sql_comment_delimiters
       assert_match %r{/\*\+ BADHINT \*/}, post_with_hint.to_sql
     end
 
+    def test_does_not_duplicate_optimizer_hints_on_merge
+      expected = "SELECT /*+ OMGHINT */ `posts`.* FROM `posts`"
```

Comment:
This is not expected query for postgresql and sqlite3 adapters.

https://buildkite.com/rails/rails/builds/62311#86409ae0-8a89-4be7-8d6b-e7b6757648da/998-1009
https://buildkite.com/rails/rails/builds/62311#888c2216-dfbf-490a-8123-eed0a3aa1912/985-994

Reply:
Existing similar test case would be of your help.

https://github.com/rails/rails/blob/d3b951f326c4facd693d5e3f92cc40ccdbf6dc1b/activerecord/test/cases/annotate_test.rb#L9-L16

Reply:
Thanks! Fixed.